windows/test-app.js

```diff
@@ -0,0 +1,258 @@
+    }, content);
+}
+
+//Binary files in React Native Test App Windows project
+const binaryExtensions = [".png", ".pfx"];
+
+/**
+ * Copy a file to given destination, replacing parts of its contents.
+ * @param srcPath Path to a file to be copied.
+ * @param destPath Destination path.
+ * @param replacements: e.g. {'TextToBeReplaced': 'Replacement'}
+ */
+function copyAndReplace(
+    srcPath,
+    destPath,
+    relativeDestPath,
+    replacements = {}
+) {
+    const fullDestPath = path.join(destPath, relativeDestPath);
+    if (fs.lstatSync(srcPath).isDirectory()) {
+        if (!fs.existsSync(fullDestPath)) {
+            fs.mkdirSync(fullDestPath);
+        }
+        return;
+    }
+
+    const extension = path.extname(srcPath);
+    if (binaryExtensions.indexOf(extension) !== -1) {
+        // Binary file
+        fs.copyFile(srcPath, fullDestPath, (err) => {
+            if (err) {
+                throw err;
+            }
+        });
+    } else {
+        // Text file
+        const srcPermissions = fs.statSync(srcPath).mode;
+        let content = resolveContents(srcPath, replacements);
+        fs.writeFile(
+            fullDestPath,
+            content,
+            {
+                encoding: "utf8",
+                mode: srcPermissions,
+            },
+            (err) => {
+                if (err) {
+                    throw err;
+                }
+            }
+        );
+    }
+}
+
+function copyAndReplaceAll(
+    srcPath,
+    destPath,
+    relativeDestDir,
+    replacements = {}
+) {
+    return Promise.all(
+        walk(srcPath).map((absoluteSrcFilePath) => {
+            const filename = path.relative(srcPath, absoluteSrcFilePath);
+            const relativeDestPath = path.join(relativeDestDir, filename);
+            return copyAndReplace(
+                absoluteSrcFilePath,
+                destPath,
+                relativeDestPath,
+                replacements
+            );
+        })
+    );
+}
+
+const cppSourceFilesExtensions = [".h", ".cpp", ".idl", ".xaml"];
+
+function linkSourceFiles(srcPath, destPath, relativeDestDir) {
+    return Promise.all(
+        walk(srcPath).map((absoluteSrcFilePath) => {
+            if (
+                cppSourceFilesExtensions.indexOf(path.extname(absoluteSrcFilePath)) !=
+                -1
+            ) {
+                const filename = path.relative(srcPath, absoluteSrcFilePath);
+                const fullDestPath = path.join(destPath, relativeDestDir, filename);
+
+                //Avoid creating links when they already exist
+                if (!fs.existsSync(fullDestPath)) {
+                    fs.link(absoluteSrcFilePath, fullDestPath, (err) => {
+                        if (err) {
+                            throw err;
+                        }
+                    });
+                }
+            }
+        })
+    );
+}
+
+async function copyProjectTemplateAndReplace(
```

Comment:
We don't call `await` here, so we don't need `async`:
```suggestion
function copyProjectTemplateAndReplace(
```